### DISCUSSION TITLE
Potential abuse of embedding external links

### DISCUSSION THREAD

**Discussion by seboslaw** [bluesky-social/atproto#1698] [Bluesky Client App]
Hey guys,

I see 2 problems with the current way external (link/url) embeddings are implemented right now. AFAIK, if the user types in a link while creating a post and then hits the "Add link card..." the client sends a request to the backend, which in turn scrapes the pages, extracts title, description and thumb and returns that information to the client. The client then uploads that information (with the post) to the Bluesky backend when the user hits the "Post" button. 

1. The fact that the API accepts title, description and thumb (along with the uri) will let any 3rd party developer give the ability to create rogue previews within the bluesky app. As a test I created a Post with an uri linking to a news article about a sea lion escaping from the NY zoo. But the title and description I sent to the backend were totally made up. See: https://bsky.app/profile/seabass.bsky.social/post/3kan6m5hxew2d 

As a result the "preview" within the Bluesky app now displays incorrect information. The problem is that a lot of users just look at the preview and don't click on the actual link to read the actual article. In times of elections this could be a real problem.

2. I'm not 100% it's completely legal from a copyright perspective to let the user "post" an "under-the-hood" scraped image (which has copyright protection) to the bluesky backend.

Since you're running the link scraping through the backend anyways, why not keep all the data there and only send a reference to the data set (title, description, thumb) to client? Then - when displaying - the client could load the data through the reference. 

Cheers,

Sebastian

**Comment by snarfed** [comment]
Yes! Previously asked here: https://github.com/bluesky-social/atproto/discussions/1304 . Also related: https://github.com/bluesky-social/atproto/discussions/1595 , https://github.com/qwell/bsky-exploits#exploit-list , https://discord.com/channels/1097580399187738645/1097580399187738648

The broad answer here is that in an open, (eventually) federated system like Bluesky, there are multiple tiers of independent services built and run by different people and organizations - client apps, PDSes, labelers, BGSes, AppViews - and it's not clear that any of them are the single "right" place to generate or render link previews, even PDSes.

Fwiw the fediverse has grappled with this for a while now. They decided not to materialize link previews at all; they instead fetch them on demand everywhere they need them, which unfortunately results in small DDoSes to sites that are linked in popular posts: https://github.com/mastodon/mastodon/issues/4486, https://github.com/mastodon/mastodon/issues/23662

**Reply by seboslaw** [reply]
oh, alright. Looks like I'm late to the party :) Thanks for the links and the explanation!
So from what I read in the #1595 ticket the plan is to mitigate the problem of fake link cards through labelling, correct?

**Reply by snarfed** [reply]
Not sure if it's a concrete plan yet, but labeling is definitely one good approach here, yes!

**Comment by Charlotte-br560** [comment]


**Reply by DavidBuchanan314** [reply]
^ this comment was generated by an LLM

**Reply by surfdude29** [reply]
I thought it seemed a bit strange... profile def looks rather spammy too
